kernel/kernel-api/src/test/scala/io/delta/kernel/internal/snapshot/LogSegmentSuite.scala

```diff
@@ -261,6 +268,36 @@ class LogSegmentSuite extends AnyFunSuite with MockFileSystemClientUtils {
     assert(exMsg === "Delta versions must be contiguous: [11, 13]")
   }
 
+  test("constructor -- delta commit files (JSON) outside of log path") {
+    // Create delta commit file lists from a different table.
+    val ex = intercept[RuntimeException] {
+      new LogSegment(
+        logPath,
+        12,
+        anotherTableDeltasFs11To12List,
+        Collections.emptyList(),
+        checkpointFs10List,
+        Optional.empty(),
+        1)
+    }
+    assert(ex.getMessage.contains("doesn't belong in the transaction log"))
+  }
+
+  test("constructor -- checkpoint files (parquet) outside of log path") {
+    // Create checkpoint file lists from a different table.
+    val ex = intercept[RuntimeException] {
+      new LogSegment(
+        logPath,
+        12,
+        deltasFs11To12List,
+        Collections.emptyList(),
+        anotherCheckpointFs10List,
```

Comment:
same as my comment above -- move this here

also, add a test for log compactions